src/npctalk.cpp

```diff
@@ -4047,9 +4066,15 @@ talk_effect_fun_t::func f_bulk_trade_accept( const JsonObject &jo, std::string_v
         if( tmp.count_by_charges() ) {
             seller->use_charges( d.cur_item, seller_has );
         } else {
-            seller->use_amount( d.cur_item, seller_has );
+            seller->use_amount( d.cur_item, seller_has_loose );
+            seller->remove_items_with( is_canned_item );
+        }
+        if( seller_cans.size() != seller_has ) {
```

Comment:
emscripten build is complaining about this: 

```
src/npctalk.cpp:4072:32: error: comparison of integers of different signs: 'size_type' (aka 'unsigned long') and 'int' [-Werror,-Wsign-compare]
 4072 |         if( seller_cans.size() != seller_has ) {
      |             ~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~
```

probably `seller_has` should be a `size_t`

Reply:
Yep I fixed that